From 446fd2a2bf04154b10e03638bd448bdba2bb732f Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sat, 10 Aug 2024 02:52:23 +0800
Subject: [PATCH] Revert "Add -export_dynamic flag for AppleDynamicLinker"

This reverts commit dfd22db4be6bfc0e64272479b51bbf314db04ac2.
---
 mesonbuild/linkers/linkers.py |  3 ---
 unittests/darwintests.py      | 12 ------------
 2 files changed, 15 deletions(-)

diff --git mesonbuild/linkers/linkers.py mesonbuild/linkers/linkers.py
index 0b8927359..103d368db 100644
--- mesonbuild/linkers/linkers.py
+++ mesonbuild/linkers/linkers.py
@@ -839,9 +839,6 @@ class AppleDynamicLinker(PosixDynamicLinkerMixin, DynamicLinker):
     def get_thinlto_cache_args(self, path: str) -> T.List[str]:
         return ["-Wl,-cache_path_lto," + path]
 
-    def export_dynamic_args(self, env: 'Environment') -> T.List[str]:
-        return self._apply_prefix('-export_dynamic')
-
 
 class LLVMLD64DynamicLinker(AppleDynamicLinker):
 
diff --git unittests/darwintests.py unittests/darwintests.py
index afc663a57..5739beca5 100644
--- unittests/darwintests.py
+++ unittests/darwintests.py
@@ -81,18 +81,6 @@ class DarwinTests(BasePlatformTests):
         self.build()
         self.run_tests()
 
-    def test_apple_lto_export_dynamic(self):
-        '''
-        Tests that -Wl,-export_dynamic is correctly added, when export_dynamic: true is set.
-        On macOS, this is relevant for LTO builds only.
-        '''
-        testdir = os.path.join(self.common_test_dir, '148 shared module resolving symbol in executable')
-        # Ensure that it builds even with LTO enabled
-        env = {'CFLAGS': '-flto'}
-        self.init(testdir, override_envvars=env)
-        self.build()
-        self.run_tests()
-
     def _get_darwin_versions(self, fname):
         fname = os.path.join(self.builddir, fname)
         out = subprocess.check_output(['otool', '-L', fname], universal_newlines=True)
